src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx

```diff
@@ -178,6 +182,20 @@ export function QuestDetailsWrapper({
       loading: val.isLoading || val.isFetching
     })) ?? []
 
+  useEffect(() => {
```

Comment:
Might be able to break these useEffects into hooks to make this a bit more readable/maintainable, maybe something like:

```ts
import { useEffect } from 'react';

// Custom hook for tracking events
function useTrackEvent(selectedQuestId: number | null) {
  useEffect(() => {
    if (selectedQuestId !== null) {
      window.api.trackEvent({
        event: 'Quest Viewed',
        properties: { quest: { id: selectedQuestId.toString() } },
      });
    }
  }, [selectedQuestId]);
}

// Custom hook for sync interval
function useSyncInterval(projectId: string, selectedQuestId: number | null, invalidateQuery: () => void) {
  useEffect(() => {
    const syncTimer = setInterval(async () => {
      await window.api.syncPlaySession(projectId, 'hyperplay');
      // allow for some time before read
      await wait(5000);
      await invalidateQuery();
      resetSessionStartedTime();
    }, 1000 * 60);

    return () => {
      clearInterval(syncTimer);
    };
  }, [projectId, selectedQuestId, invalidateQuery]);
}
```

Reply:
made the refactor 👍